### DISCUSSION TITLE
vtk modification to Nuitka for vedo compilation

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#693] [General]
Hi Marco, 
This is just a quick note to let you know about the above. I attempted to package the program I have been working on to an executable, so that I could hand the software to a colleague for easy use. After that compilation process, I kept encountering a problem with vtk (apparently, the python iteration is not packaged very well). 

I raised an issue with the Nuitka developers and they looked at it, and realised it needed particular attention and a modification of their code base enough to incorporate it into a following release. When they examined the interaction of vedo with Nuitka (which wasn't optimal), they made modifications to incorporate that also. The long and short of it is that now vedo should comfortably compile in Nuitka! 

I have included a link to the github issue below:
[Compiled program failing with 'vtk.util' ModuleNotFoundError](https://github.com/Nuitka/Nuitka/issues/1718)

I just wanted to inform you of this, so you were aware. The release is not out yet (kay said it was release 1.1), so I haven't tested it but I need this for my project, so if there are issues, I bring it back up with the Nuitka devs. 

Have a nice day!
j

**Comment by marcomusy** [comment]
Thanks @jkissi I will definitely have a look at Nuitka, I was not aware of it and it looks interesting!
the `vedo.backend` is indeed bloated and would need some restructuring... but never found the right moment to do it...

**Reply by jkissi** [reply]
Well I think they've been able to deal work around it quite easily. Just from reading some of the issues on their github, it seems like there are some really bloated python libraries out there. They barely talked about it in the thread, so I dont think it was a problem

**Comment by jkissi** [comment]
Also, when you do finally have time to look at it, maybe it would be a good idea to talk to the Nuitka devs to find out what kind of improvements would work best for compilation to executable. I personally think very highly of your work here, and it would great if vedo reliably compiled to executable format because this package could then be very easily incorporated into practical software developments. 

Just a thought, hope you're having a good one!!
j
